backend/routes/appointments/viewAppointmentsRouter.js

```diff
@@ -51,10 +51,52 @@ module.exports = ({ prisonApi, whereaboutsApi, logError }) => async (req, res) =
       court: appointment.court,
     }))
 
+  const addBookingIdAndHearingTypesToVLBAppointments = appointment => {
+    const appCopy = { ...appointment }
+    if (appointment.appointmentTypeCode === 'VLB') {
+      videoLinkAppointments.forEach(vlAppointment => {
+        if (vlAppointment.appointmentId === appointment.id) {
+          appCopy.hearingType = vlAppointment.hearingType
+          appCopy.bookingId = vlAppointment.bookingId
+        }
+      })
+    }
+    return appCopy
+  }
+
+  const addDisplayTimesForVLBAppointments = (appointment, index, appointmentsArray) => {
+    const appCopy = { ...appointment }
+    const preAppointment = appointmentsArray.find(
+      otherAppointment => otherAppointment.bookingId === appointment.bookingId && otherAppointment.hearingType === 'PRE'
+    )
+    const postAppointment = appointmentsArray.find(
+      otherAppointment =>
+        otherAppointment.bookingId === appointment.bookingId && otherAppointment.hearingType === 'POST'
+    )
+    if (preAppointment) {
+      appCopy.displayStartTime = preAppointment.startTime
+    }
+    if (postAppointment) {
+      appCopy.displayEndTime = postAppointment.endTime
+    }
+    return appCopy
```

Comment:
I think it is best to avoid adding new variables if possible. Could you not return a structure with the start/endTimes reassigned, e.g.
"
return {
  ...appointment,
  startTime: displayStartTime || startTime,
  endTime: displayEndTime || endTime,
}
"

..you would then not need to do the "displayEndTime || endTime" further down, which adds some confusion.